### PR TITLE
[Carry 17059] allow filtering containers by any status

### PR DESCRIPTION
**Pull request by calavera**
Carry #17059.

I added a test to make sure this feature works as expected.

/cc @jfrazelle because she left a comment in that test that I don't think applies anymore.

### CODE REVIEW

File: integration-cli/docker_cli_ps_test.go [hunk 1]
```diff
@@ -245,6 +244,18 @@ func (s *DockerSuite) TestPsListContainersFilterStatus(c *check.C) {
...
 	}
 
+	// pause running container
+	out, _ = dockerCmd(c, "run", "-itd", "busybox")
+	pausedID := strings.TrimSpace(out)
+	dockerCmd(c, "pause", pausedID)
+	// make sure the container is unpaused to let the daemon stop it properly
+	defer func() { dockerCmd(c, "unpause", pausedID) }()
+
+	out, _ = dockerCmd(c, "ps", "-q", "--filter=status=paused")
```
Comment (tiborvass) [inline integration-cli/docker_cli_ps_test.go]: if you use `--no-trunc` you don't need truncate with `[:12]` and it's more robust

**Comment by tiborvass** [comment]
apart from small nit, lgtm

**Comment by vdemeester** [comment]
@calavera I wonder about the test. It already works with the current master (even 1.8.3) : 

```bash
λ docker run -itd busybox 
0471b3abc8ad5f2edfd22ad62c1bb2515154e6f5230260ac1e477bc751f52672
λ docker pause 0471b3abc8ad5f2edfd22ad62c1bb2515154e6f5230260ac1e477bc751f52672
0471b3abc8ad5f2edfd22ad62c1bb2515154e6f5230260ac1e477bc751f52672
λ docker ps -a 
CONTAINER ID        IMAGE                COMMAND                  CREATED             STATUS                    PORTS               NAMES
0471b3abc8ad        busybox              "sh"                     9 seconds ago       Up 8 seconds (Paused)                         happy_turing
a1b1c4074925        vdemeester/spotify   "spotify -stylesheet="   34 hours ago        Exited (0) 26 hours ago                       spotify
λ docker ps -q --filter=status=paused
0471b3abc8ad
```

**Comment by LK4D4** [comment]
@calavera what can you say about tests? :)

**Comment by calavera** [comment]
It looks like "paused" is considered a "running" status. So the only state that won't be listed without this patch is "dead". It's pretty hard to get a container in that state in the tests. I could create a json configuration with that state on disk and see what happens, but it don't look very nice.

**Comment by LK4D4** [comment]
@calavera need rebase.

**Comment by calavera** [comment]
rebased

**Comment by LK4D4** [comment]
LGTM

**Comment by ozlerhakan** [comment]
how can I test `docker ps -f status=dead` in the latest rc ? @calavera  @vdemeester

**Comment by thaJeztah** [comment]
status "dead" is an exceptional case, so hard to test

**Comment by ozlerhakan** [comment]
hi @thaJeztah ,thank you for the quick reply. maybe it is also hard to explain i, but just out of curiosity,  in which stuation can one container be in a dead state?

**Comment by thaJeztah** [comment]
Usually that is because (for example) a container was attempted to be removed, but something failed during that process; basically the container is then defunct, and will be removed/cleaned up on daemon restart.

**Comment by ozlerhakan** [comment]
thanks @thaJeztah

**Pull request merged**